go/teams/chain.go

```diff
@@ -131,16 +131,22 @@ func (t TeamSigChainState) GetUserLogPoint(user keybase1.UserVersion) *keybase1.
 	return &tmp
 }
 
-// GetLastUserLogPointWithPredicate gets the last user logpoint in the series for which the given
-// predicate is true.
-func (t TeamSigChainState) GetLastUserLogPointWithPredicate(user keybase1.UserVersion, f func(keybase1.UserLogPoint) bool) *keybase1.UserLogPoint {
+// GetLastChangeLogPointWithPredicates gets the first user logpoint in the series at which
+// the second predicate is true AFTER the last user logpoint at which the first predicate is true.
+func (t TeamSigChainState) GetLastChangeLogPointWithPredicates(user keybase1.UserVersion,
```

Comment:
Do you think it's possible do to this with one predicate that stores some state?  Otherwise, maybe we should get rid of this function and just implement what we want directly.  It doesn't seem like we're ever going to have a second use case for this particular function.

Reply:
Seems like from above, that `f1(x) = !f2(x)` in all use cases, so we can simplify this...

Reply:
ahh yes yes. i wasn't actually sure if this would be strictly true based on underlying data and non-obvious uses. but yes, it definitely seems to be the case if this usecase is executed in isolate. so... lovely! i will simplify this.